crates/driver/src/boundary/mempool.rs

```diff
@@ -42,6 +42,12 @@ pub struct Config {
 #[derive(Debug, Clone)]
 pub enum Kind {
     /// The public mempool of the [`Ethereum`] node.
+    ///
+    /// Don't submit transactions with high revert risk (i.e. transactions
+    /// that interact with on-chain AMMs) to the public mempool.
+    /// This can be enabled to avoid MEV when private transaction
+    /// submission strategies are available. If private submission strategies
+    /// are not available, revert protection is always disabled.
```

Comment:
I think this comment should not be on the enum variant because this invariant is not guaranteed by the `Kind` itself via a constructor or something like that.

Reply:
Moved to `RevertProtection`